content/blog/2022/2022-08-25-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,318 @@
+
+Here is an example of what the third argument looks like.
+```javascript
+{
+   scheduled: false,
+   timezone: "America/Sao_Paulo"
+}
+```
+the scheduled option here is a boolean to indicate whether the task created is scheduled (default is true).
+
+while the timezone option can be used to provide consistency in your scheduled jobs, by forcing them to adhere to a specific timezone or area.
+
+## Setting up Your Node.Js Application
+Installing dependencies and creating our node application.
+
+To begin, open your terminal and create a new folder.
+```
+mkdir node-cron-demo
+```
+
+Next change into the new project's directory
+```
+cd node-cron-demo
+```
+You will need to create a file index.js here, this is where we'll be writing all our codes.
+```
+touch index.js
+```
+Run the command below to initialize the project. This will generate a `package.json` file which can be used to keep track of all dependencies installed in our project.
+```
+npm init -y
+```
+
+Next, we will install `node-cron` and other modules used later in this article.
+```
+npm install node-cron node-mailer 
+```
+After installation, the dependencies can be required in our node applications
+
+## What We Will Be Building
+To demonstrate the functionality of the node-cron library, we will build `4` sample applications using node.js.
+
+### 1. Scheduling a Simple Task with Node Cron.
+Any task of your choosing can be automated and run at a specific time using cron job schedulers. In this section, we'll write a simple function that logs to the terminal at our specified time.
+
+Input the following code into the `index.js` file to create our simple task scheduler:
+```javascript
+const cron = require("node-cron");
+const express = require("express");
+
+const app = express();
+
+cron.schedule("*/15 * * * * *", function () {
+  console.log("---------------------");
+  console.log("running a task every 15 seconds");
+  //replace simple tasks
+});
+
+app.listen(3000, () => {
+  console.log("Application Listening.....");
+});
+```
+In the code block above we are making a simple log to the application's terminal.
+
+Run `node index.js` in the terminal.
+
+`output`
+```
+Application Listening.....
+---------------------
+running a task every 15 second
+```
+
+
+### 2. Scheduling Email Using Node Cron
+Emailing is a common feature of modern applications. Cron jobs can be used to accomplish this. For instance, a job schedule can be set to automatically send users an email each month with the most recent information from a blog or a product.
+
+Here we are using google's email service provider `gmail`, if you have a Gmail account insert it in the code below to test out our newly created email scheduler.
+
+Note: To use node-mailer with Gmail, you must first create an [app password](https://support.google.com/mail/answer/185833?hl=en-GB) for Gmail to allow third-party access.
+
+Set up your Gmail app password following these steps:
```

Comment:
Nice. This will make it really easy for readers to follow along

Reply:
This is fine right?